scripts/zkai_native_d128_verifier_execution_compression_budget_gate.py

```
@@ -0,0 +1,679 @@
+    cases = require_list(payload.get("mutation_cases"), "mutation cases")
+    names = tuple(require_str(require_dict(case, "mutation case").get("name"), "mutation case name") for case in cases)
+    if names != MUTATION_NAMES:
+        raise CompressionBudgetGateError("mutation inventory drift")
+    checked = require_int(payload.get("mutations_checked"), "mutations_checked")
+    rejected = require_int(payload.get("mutations_rejected"), "mutations_rejected")
+    if checked != len(MUTATION_NAMES) or rejected != len(MUTATION_NAMES):
+        raise CompressionBudgetGateError("mutation count drift")
+    if payload.get("all_mutations_rejected") is not True:
+        raise CompressionBudgetGateError("mutation rejection summary drift")
+    for case in cases:
+        case = require_dict(case, "mutation case")
```

Ruff (0.15.12)

[warning] `for` loop variable `case` overwritten by assignment target

(PLW2901)